### PR TITLE
refactor(desktop): keep the durable event shape in the transcript model

### LINKED ISSUE #1075: OpenSeek transcript renders same-step activity before assistant prose
## Summary

OpenSeek renders reasoning/tool activity from a durable assistant step before that step's user-facing prose.

For an assistant event containing prose followed by a tool call, the transcript currently appears as:

```text
activity / tool call
assistant prose
```

Expected:

```text
assistant prose
activity / tool call
```

## Minimal reproduction

Use one durable OpenSeek assistant sequence with these visible rows:

```text
Assistant("I will inspect the project.", sequence=2)
Tool(name="read", arguments="{}", sequence=2)
```

A temporary white-box test against `TranscriptRenderer::stream_children` consistently produced:

```text
actual:
["activity\0head", "row\0s2\0o0"]

expected:
["row\0s2\0o0", "activity\0after\0s2\0o0"]
```

The test failed identically on two consecutive runs. A DOM-level reproduction through the production `transcript_view` also placed the activity markup before the prose (`activity_at=511`, `assistant_at=926`) on two runs.

As a control, changing only the transcript source from `OpenSeek` to `Codex` restored the expected prose-before-activity order.

## Root cause

The durable session projector preserves the assistant payload structure in `desktop/frontend/transcript/sessions.mbt:296-339`. In particular, tool calls are appended after visible assistant prose (`:322-339`).

The OpenSeek-only durable-step folding branch in `desktop/frontend/transcript/component/rendering.mbt:197-225` then:

1. collects every `Reasoning` and `Tool` row with the same durable sequence;
2. inserts the combined activity node at `:208-212`;
3. inserts assistant prose afterward at `:217-225`.

The comment at `:214-216` explicitly describes this activity-first behavior. The assertion in `desktop/frontend/transcript/component/view_tests.mbt:258-262` currently locks it in as expected output.

This rules out CSS, Rabbita keyed reconciliation, session projection, and streaming timing as the cause: the keyed child map itself is already in the wrong display order for OpenSeek.

## Suggested fix

Within the same durable-step folding branch, render non-activity prose before inserting the combined activity node. Preserve the activity key derived from the pre-step `left_boundary` so that the live-to-durable handoff can continue reusing the browser-owned `<details>` node even when it moves after the newly committed prose.

Update the existing display-order assertion and add a regression test for an assistant event containing both non-empty `content` and `tool_calls`.

### PR DESCRIPTION
**Pull request by tonyfettes**
Closes #1075. Supersedes #1163.

## Summary

The desktop transcript projector now keeps the durable event's own shape instead of flattening one Assistant event into Reasoning / Assistant / Tool rows that the renderer re-grouped by sequence. One durable event projects to at most one `TranscriptItem`, and `stream_children` is a plain per-item keyed loop.

- `ItemKind` carries each kind's text: `User(String) | Assistant(reasoning~, prose~, calls~, replay~, finished~) | Answer(String) | Failure(String) | Notice(label~, brief~, is_error~, body~) | Summary(String)`. A part that may be absent (a response's prose, a notice's body) is an option, never an empty string; `TranscriptItem::text()` is the one accessor for "the item's text, if any".
- `ToolCall { call_id, name, arguments, outcome }` with `ToolOutcome::Pending(output~)` (Codex may stream output before the item completes) and `ToolOutcome::Done(content~, is_error~, brief~)`.
- `apply_session_item`: an Assistant event is one item; a `ToolResult` fills the pending call inside the response that made it; a `Finished` terminal matching the last response's prose marks it `finished` (the response keeps its own stamp and sequence, so the settled bubble stays the same keyed node); other terminals are `Failure`; goal markers and runtime notices are `Notice`.
- Unmatched tool results are ignored, as `agent_session/projection.mbt` already does. Store format v1 has always persisted `tool_calls`, the engine appends the response before any result, and a scan of every local store (172 sessions, 11797 results) found no unmatched result, so the orphan row had no producer.
- Renderer: row identity is `s<sequence>`; the step ordinal is a render-time count of non-replay responses; the live preview is one `live\0<run>` node rendered by the same `assistant_view` a settled response gets. `durable_step_ordinals`, `left_boundary`, `activity_after` and the `rendered_prose` special case are gone.
- A response renders as one `div.step` whose parts keep the model's order: thought fold, prose, tool rows (the viz renderer's `assistant_card` order). `.step` reuses the stream grid so spacing is unchanged. A notice renders through the same result-row view a tool result uses, without pretending to be a tool call.
- Consumers follow the model: overview `entries` (reply = first item with text among responses, answers and failures; failed = any errored call or blocked goal), `plan.classify/latest/fold_progress` take a `ToolCall`, Codex maps each app-server item to its own response-shaped item.

## Behaviour changes worth a look

- Codex adjacent reasoning and tool items are no longer merged into one card; each is its own step (stream gap instead of the card's 6px).
- The overview reply skips responses without prose (a thought-only or tool-only response is not an answer yet).
- The live preview node is replaced by the durable node at commit instead of being relocated; the fold's `open` attribute flipped at that moment already.

## Tests

- `moon check --target js` / `--target native`: clean.
- `moon test --target js`: 3182 passed. `moon test --target native`: 3410 passed.
- Playwright: new case "a model step shows its thought, then its prose, then its tool rows" asserts the step's DOM order (js has no SSR, so the layout is covered end to end). The overview fixture gained the assistant call its error result was answering. The rest of the suite passes locally except "sidebar menu dismissal and pending selection follow the clicked row", which fails identically on `main` in this environment.
- js-only `pkg.generated.mbti` synced from `_build`; the transcript interface also drops two entries that were already stale (`RuntimeUpdate`, `parse_runtime_update`).

## Not verified

- No run in the desktop app itself; the live-to-durable handoff and the Codex spacing deserve a look there.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01DYEUEQDanXg32aFCFMXH8a

### CODE REVIEW

File: desktop/frontend/transcript/component/rendering.mbt [hunk 1]
```diff
@@ -201,114 +132,65 @@ fn TranscriptRenderer::stream_children(
...
+  // checkpoint replay is one. Codex and legacy items carry no ordinal.
+  let mut steps = 0
+  for index, row in visible {
+    let item = row.item
+    let step_label = if self.input.source is OpenSeek &&
+      item.kind is Assistant(replay=false, ..) {
+      steps += 1
+      "#\{steps}"
+    } else {
+      ""
```
[the remark this answers is not in the document]
Comment (tonyfettes) [reply desktop/frontend/transcript/component/rendering.mbt]: Will be refactored in another PR
